### ISSUE TITLE
3.14 field annotation not covered

### ISSUE THREAD

**Issue by nickdrozd** [coveragepy/coveragepy#1908]
The following snippet gets counted as 100% coverage under 3.13 and earlier versions, but under 3.14 it is counted as having a missing line:

```python
class X:
    x: int  # <-- not covered under 3.14
```

```
python3.14 -m coverage run asdf.py && python3.14 -m coverage combine --quiet && python3.14 -m coverage report --fail-under 100
Name      Stmts   Miss Branch BrPart  Cover
-------------------------------------------
asdf.py       2      1      0      0    50%
-------------------------------------------
TOTAL         2      1      0      0    50%
Coverage failure: total of 50 is less than fail-under=100
```

Adding `from __future__ import annotations` causes everything to be counted:

```python
from __future__ import annotations

class X:
    x: int  # <-- covered
```

Could have something to do with https://docs.python.org/3.14/whatsnew/3.14.html#whatsnew314-pep649?

**Comment by nedbat** [comment]
This definitely is because of deferred annotations.  I'm not sure what to do here.  The fact is there is code that has not been run. But also, it is likely never to be run, and it's fine that it isn't run.

But it can have errors.  This will be an error if you run it in 3.13:
```python
class X:
    x: inttttt
    y = 1
```
```
% python3.13 att.py
Traceback (most recent call last):
  File "/Users/ned/coverage/trunk/att.py", line 1, in <module>
    class X:
        x: inttttt
        y = 1
  File "/Users/ned/coverage/trunk/att.py", line 2, in X
    x: inttttt
       ^^^^^^^
NameError: name 'inttttt' is not defined
```
In 3.14, it won't error, but it will if you try to get the annotation:
```python
class X:
    x: inttttt
    y = 1

from annotationlib import get_annotations, Format
get_annotations(X, format=Format.VALUE)
```
```
% python3.14 att.py
Traceback (most recent call last):
  File "/Users/ned/coverage/trunk/att.py", line 7, in <module>
    get_annotations(X, format=Format.VALUE)
    ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/pyenv/pyenv/versions/3.14.0a3/lib/python3.14/annotationlib.py", line 707, in get_annotations
    ann = _get_dunder_annotations(obj)
  File "/usr/local/pyenv/pyenv/versions/3.14.0a3/lib/python3.14/annotationlib.py", line 842, in _get_dunder_annotations
    ann = _BASE_GET_ANNOTATIONS(obj)
  File "/Users/ned/coverage/trunk/att.py", line 2, in __annotate__
    x: inttttt
       ^^^^^^^
NameError: name 'inttttt' is not defined
```

**Comment by nedbat** [comment]
btw, I also opened a discussion thread about this: https://discuss.python.org/t/how-should-coverage-py-treat-deferred-annotations/74990

**Comment by nickdrozd** [comment]
As a user, my feeling is that if the field annotation is considered covered in 3.13 with `from __future__ import annotations`, then it should also be considered covered under 3.14 without the import.

Here is a naive suggestion. I assume there is a line somewhere to the effect of:

```python
if has_future_annotations_import():
    field_annotation.count_as_covered()
```

Maybe that could be changed to:

```python
if has_future_annotations_import() or PY_VER_314_PLUS:
    field_annotation.count_as_covered()
```

**Comment by nedbat** [comment]
All of the annotations are put into code objects called `__annotate__`, so I can ignore those when looking for executable lines, and the problem is fixed in commit 2d048353b2d1c62f3eee9d82d7ac055229e8e791.

**Comment by nickdrozd** [comment]
Tested; works. Thanks!

**Comment by nedbat** [comment]
This is now released as part of [coverage 7.6.10](https://pypi.org/project/coverage/7.6.10).
